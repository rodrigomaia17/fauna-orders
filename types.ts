export interface Order {
  _id: string;
  status: string;
  shipAddress: Address;
  line: Line[];
  customer: Customer;
  totalPrice: number;
}

export interface Customer {
  name: string;
}

export interface Product {
  name: string;
  description: string;
}

export interface Line {
  product: Product;
  quantity: number;
  price: number;
  total: number;
}

export interface Address {
  zipCode: string;
  street: string;
  city: string;
  state: string;
}
