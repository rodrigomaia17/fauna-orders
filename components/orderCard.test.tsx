import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import OrderCard from './orderCard';
import { Order } from '../types';
import '@testing-library/jest-dom/extend-expect';

const order: Order = {
  _id: '',
  status: 'processing',
  shipAddress: {
    zipCode: 'test',
    street: 'test',
    city: 'test',
    state: 'test'
  },
  totalPrice: 0,
  line: [
    {
      product: {
        name: 'product',
        description: 'description'
      },
      quantity: 1,
      price: 2,
      total: 2
    }
  ],
  customer: {
    name: 'customerName'
  }
};

test('print the Order passed as param', () => {
  const { getByText, container } = render(<OrderCard order={order} />);
  expect(getByText('customerName')).toBeVisible();
  expect(container.firstChild).toMatchSnapshot();
});

describe('Collapsible behaviour', () => {
  test('start collapsed', () => {
    const { getByText } = render(<OrderCard order={order} />);
    expect(
      getByText('Ship To').parentElement.parentElement.className
    ).toContain('is-hidden');
  });

  test('expand after click', () => {
    const { container } = render(<OrderCard order={order} />);
    const box = container.firstChild as HTMLElement;
    fireEvent.click(box);
    expect(box.children[1].className).not.toContain('is-hidden');
  });
});
