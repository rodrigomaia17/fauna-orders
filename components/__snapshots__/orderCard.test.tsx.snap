// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`print the Order passed as param 1`] = `
<div
  class="jsx-733811595 box is-fullwidth"
>
  <div
    class="jsx-733811595 box-line"
  >
    <span
      class="jsx-733811595 box-line-name is-size-7-mobile"
    >
      customerName
    </span>
    <span
      class="jsx-733811595 box-line-products has-text-grey has-text-weight-light is-size-7-mobile"
    >
      1
       Products
    </span>
    <span
      class="jsx-2445842882 box-line-status is-size-7-mobile tag is-warning"
    >
      processing
    </span>
    <span
      class="jsx-733811595 box-line-price has-text-primary is-size-4 is-size-7-mobile"
    >
      0.00
    </span>
    <span
      class="jsx-733811595 icon"
    >
      <i
        class="jsx-733811595 material-icons"
      >
        keyboard_arrow_down
      </i>
    </span>
  </div>
  <div
    class="jsx-733811595 card-content is-hidden"
  >
    <div
      class="jsx-733811595 content"
    >
      <div
        class="container"
      >
        <p>
          Products 
        </p>
        <div
          class="table-container"
        >
          <table
            class="table is-narrow is-hoverable is-striped"
          >
            <thead>
              <tr>
                <th>
                  Name
                </th>
                <th>
                  Description
                </th>
                <th>
                  Quantity
                </th>
                <th>
                  Price
                </th>
                <th>
                  Subtotal
                </th>
              </tr>
            </thead>
            <tbody>
              <tr
                class="has-text-weight-light"
              >
                <td>
                  product
                </td>
                <td>
                  description
                </td>
                <td>
                  1
                </td>
                <td>
                  2.00
                </td>
                <td>
                  2.00
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <hr
        class="jsx-733811595 hr"
      />
      <p
        class="jsx-733811595"
      >
        Ship To
      </p>
      <p
        class="jsx-733811595 has-text-weight-light"
        style="margin-bottom: 0px;"
      >
        test
      </p>
      <p
        class="jsx-733811595 has-text-weight-light"
      >
        test
         
        test
         
        test
      </p>
    </div>
  </div>
</div>
`;
